modules/gke/main.tf

```diff
@@ -109,3 +113,216 @@ resource "google_service_account_iam_binding" "workload_identity" {
+  }
+}
+
+resource "helm_release" "openebs" {
+  count = var.install_openebs ? 1 : 0
+
+  name       = "openebs"
+  namespace  = var.openebs_namespace
+  repository = "https://openebs.github.io/openebs"
+  chart      = "openebs"
+  version    = var.openebs_version
+
+  set {
+    name  = "engines.replicated.mayastor.enabled"
+    value = "false"
+  }
+
+  # Unable to continue with install: CustomResourceDefinition "volumesnapshotclasses.snapshot.storage.k8s.io"
+  # in namespace "" exists and cannot be imported into the current release
+  # https://github.com/openebs/website/pull/506
+  set {
+    name  = "openebs-crds.csi.volumeSnapshots.enabled"
+    value = "false"
+  }
+
+
+  depends_on = [
+    google_container_node_pool.primary_nodes,
+    kubernetes_namespace.openebs
+  ]
+}
+
+resource "kubernetes_namespace" "disk_setup" {
+  count = var.enable_disk_setup ? 1 : 0
+
+  metadata {
+    name = "disk-setup"
+    labels = {
+      "app.kubernetes.io/managed-by" = "terraform"
+      "app.kubernetes.io/part-of"    = "materialize"
+    }
+  }
+
+  depends_on = [
+    google_container_node_pool.primary_nodes
+  ]
+}
+
+# Create a ConfigMap containing the disk setup script
+resource "kubernetes_config_map" "disk_setup_script" {
+  count = var.enable_disk_setup ? 1 : 0
+
+  metadata {
+    name      = "disk-setup-script"
+    namespace = kubernetes_namespace.disk_setup[0].metadata[0].name
+  }
+
+  data = {
+    "bootstrap.sh" = file("${path.module}/bootstrap.sh")
+  }
+
+  depends_on = [
+    kubernetes_namespace.disk_setup
+  ]
+}
+
+resource "kubernetes_daemonset" "disk_setup" {
+  count = var.enable_disk_setup ? 1 : 0
+
+  metadata {
+    name      = "disk-setup"
+    namespace = kubernetes_namespace.disk_setup[0].metadata[0].name
+    labels = {
+      "app.kubernetes.io/managed-by" = "terraform"
+      "app.kubernetes.io/part-of"    = "materialize"
+      "app"                          = "disk-setup"
+    }
+  }
+
+  spec {
+    selector {
+      match_labels = {
+        app = "disk-setup"
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          app = "disk-setup"
+        }
+      }
+
+      spec {
+        security_context {
+          run_as_non_root = false
+          run_as_user     = 0
+          fs_group        = 0
+          seccomp_profile {
+            type = "RuntimeDefault"
+          }
+        }
+
+        affinity {
+          node_affinity {
+            required_during_scheduling_ignored_during_execution {
+              node_selector_term {
+                match_expressions {
+                  key      = "materialize.cloud/disk"
+                  operator = "In"
+                  values   = ["true"]
+                }
+              }
+            }
+          }
+        }
+
+        # Use host network and PID namespace
+        host_network = true
+        host_pid     = true
+
+        init_container {
+          name  = "disk-setup"
+          image = "debian:bullseye-20250407-slim"
```

Comment:
We should probably bake LVM into an image, or find one that comes with it.

We may be able to simplify this module quite a bit if we also bake the bootstrap script into the image.